Assignment3Test.java

```diff
@@ -0,0 +1,17 @@
+import org.junit.jupiter.api.Test;
+
+import java.util.Arrays;
+import java.util.HashSet;
+
+import static org.junit.jupiter.api.Assertions.*;
+
+class Assignment3Test {
+    char[][] grid = {{'a', 'a', 'r'}, {'t', 'c', 'd'}};
+
+    @Test
+    void findAllWordsTest() {
+        assertEquals(new HashSet<>(new Assignment3().findAllWords(2, 3, grid, new MyDictionary())),
```

Comment:
Usual practice is to create tested object in a separate method annotated with `@Before` and only use it in test.

Also: clever use of wrapping in HashSets to work around changing order of elements! There are nice [libraries](https://google.github.io/truth/) that let you handle it in a cleaner way[1], but feel free to leave it as is to not overcomplicate things.

[1] `assertThat(result).containsExactly(s1, s2, s3);`